### ISSUE TITLE
inference with jpg files

### ISSUE THREAD

**Issue by suke27** [xinntao/ESRGAN#5]
Hello, As I know GAN algorithm has problem on jpg images.
do you try your algorithm for jpg files? is there abnormal noise texture exist in output?

**Comment by rahidz** [comment]
Same! Whenever I try a jpg file the output is very “blocky”’ like there’s small but very noticeable square blocks covering the entire image. I assumed this was esrgan simply “enhancing” the jpg artifacts already present, is that accurate or does it actually create new/worse ones?

I’ve noticed slight improvement by converting the jpg to a png before I start, but it’s still not great.

**Comment by suke27** [comment]
@xinntao  for suggestion 2) finetuning the network with datasets with JPEG artifacts, so that the ESRGAN can handle them.
I try to finetune using jpg dataset for SRGAN, but not work well, also block and noise exist
but same training method for EDSR works well.

**Comment by xinntao** [comment]
@suke27 You can try to first pre-train the network with L2/L1 loss with JPEG compressed images. And then use the pre-trained model to train a GAN-based model. 

I think, L2/L1 loss is a clear and effective loss to guide the network to remove the JPEG compression artifacts (Therefore, finetuning with jpg dataset works well for EDSR). But for SRGAN, the VGG + Adv loss may be not enough to remove the JPEG artifacts. So I think first pre-training the network with JPEG compressed images to let the network learn to remove JPEG artifacts will work.

**Comment by neilthefrobot** [comment]
I was able to get around this by taking my HR training set, down sampling 4x, then converting it to .jpg with a low quality (high compression) and using that as my LR set. This way the network is seeing jpg artifacts as inputs and a non jpg artifact version as a target and it learns to convert between them. It actually worked very well too.
